app/Http/Controllers/StoreController.php

```diff
@@ -242,6 +246,29 @@ public function postCheckout()
         return js_view('store.order-create');
     }
 
+    public function putRequestNotification($product_id)
+    {
+        $user = Auth::user();
+        $product = Store\Product::findOrFail($product_id);
+
+        if ($product->inStock()) {
+            return error_popup(trans('store.product.notification-in-stock'));
+        }
+
+        $request = $product->notificationRequests()->where('user_id', $user->user_id)->first();
+
+        if ($request) {
+            $request->delete();
```

Comment:
two separate methods (`post`/`delete`, or just extra parameter) are probably better though otherwise clicking the link twice will result in noop.